tests/testing_daemon.c

```diff
@@ -0,0 +1,103 @@
+#define _GNU_SOURCE
+
+#include <stdio.h>
+#include <string.h>
+#include <stddef.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <signal.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include "config.h"
+
+#define BUFFER_LENGTH 16
+
+int get_server_socket(void)
+{
+        int sockfd;
+        struct sockaddr_un serveraddr;
+        fputs("Creating socket\n", stderr);
+        sockfd = socket(AF_UNIX, SOCK_STREAM, 0);
+        if (sockfd < 0) {
+                fputs("Fail creating socket\n", stderr);
+                return -1;
+        }
+        memset(&serveraddr, 0, sizeof(serveraddr));
+        serveraddr.sun_family = AF_UNIX;
+        strcpy(serveraddr.sun_path, SOCKET_PATH);
```

Comment:
Can we use a different socket path for test purposes? `SOCKET_PATH` points to the system-wide location used by `clr_debug_daemon.socket` and `clr_debug_daemon.service`, so I think it would be preferable to isolate test runs from what the system services are using.

Reply:
this means compile it. with a different SOCKET_PATH location. in these tests we are testing the compiled binaries then they are going to receive requests from SOCKET_PATH location if we change it in the tests we would need to change it in the bins (clr_debug_daemon and clr_debug_fuse) which means just compile the project using a different SOCKET_PATH.

Reply:
Okay, understood. It will be nice to use a different socket path for test purposes, but the code would need some refactoring to make this easier to implement.

See, for example, how some of the telemetrics-client telemd tests work: https://github.com/clearlinux/telemetrics-client/blob/master/tests/check_probd.c

Reply:
this has not being resolved yet, working on it